src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs

```diff
@@ -19,6 +19,12 @@ namespace AutoRest.CSharp.Generation.Writers
 {
     internal class LowLevelClientWriter
     {
+        internal enum CredentialKind {
```

Comment:
Feels like something that can go onto the client.

Reply:
I looked into it, but right now we get it from the configuration:

`context.Configuration.CredentialTypes.Contains("TokenCredential", StringComparer.OrdinalIgnoreCase)`

which I wasn't sure made it make sense? 

I can add a few bools to the model and push this up:

```
            bool hasKeyAuth = HasKeyAuth (context);
            bool hasTokenAuth = HasTokenAuth (context);
```

and the enum if you think that's ok.

Reply:
Maybe something to look into when we resurect the other auth PR